packages/ui-kit/src/components/Divider/Divider.tsx

```diff
@@ -0,0 +1,10 @@
+import classNames from 'classnames'
+import React from 'react'
+import componentStyles from './Divider.module.scss'
+
+export type DividerProps<T extends React.ElementType = 'hr'> = React.ComponentPropsWithoutRef<T> & { as?: T }
+
+export const Divider = <T extends React.ElementType = 'hr'>({ as, className, ...rest }: DividerProps<T>) => {
+  const Component = as || 'hr'
```

Comment:
I believe, in this particular case `||` is preferable to `??` because with `??` it will fallback to `hr` only when `as` is `undefined` or `null` only. With `||` it will fallback also for `""`, `false`, and even `0`. It just covers more falsy cases.